src/src/extra-providers.ts

```diff
@@ -8,23 +8,60 @@ import {
 
 @Injectable()
 export class SingleSpaPlatformLocation extends ɵBrowserPlatformLocation {
-  /**
-   * We could use the `inject` function from `@angular/core` which resolves
-   * dependencies from the currently active injector, but it's a feature
-   * of Angular 8+. Should be used when we drop support for older versions.
-   */
   private ngZone!: NgZone;
 
+  // This is a simple marker that helps us to ignore PopStateEvents
+  // that was not dispatched by the browser.
+  private skipNextPopState = false;
+
+  private onPopStateListeners: ((event: LocationChangeEvent) => void)[] = [];
+
+  destroy(): void {
```

Comment:
Does an instance of SingleSpaPlatformLocation ever get reused after `destroy()` is called? Or is it always reconstructed when it's remounted/rebootstrapped?

If it can get reused, we should reset `this.onPopStateListeners = []` inside of this function.